terraform-aws-observability-common/alb_source.tf

```diff
@@ -0,0 +1,66 @@
+resource "sumologic_elb_source" "this" {
+  for_each = toset(var.manage_alb_logs_source ? ["this"] : [])
+
+  category             = var.alb_logs_source_category
+  collector_id         = sumologic_collector.hosted["this"].id
+  content_type         = "AwsElbBucket"
+  cutoff_relative_time = "-1d"
+  description          = "This AwsElbBucket source is created by AWS SAM Application"
+  fields = {
+    "account"   = var.account_alias
+    "namespace" = "aws/applicationelb"
+    "region"    = data.aws_region.current.id
+  }
+  name          = "${var.account_alias}-${var.alb_logs_source_name}-${data.aws_region.current.id}"
+  paused        = false
+  scan_interval = var.scan_interval
+
+  authentication {
+    type     = "AWSRoleBasedAuthentication"
+    role_arn = aws_iam_role.sumologic_source["this"].arn
+  }
+
+  path {
+    type            = "S3BucketPathExpression"
+    bucket_name     = var.manage_alb_s3_bucket ? aws_s3_bucket.common["this"].id : var.alb_s3_bucket
+    path_expression = "*AWSLogs/*/elasticloadbalancing/${var.alb_s3_bucket_path_expression}"
```

Comment:
Can you please check the path expression. It should be based on condition. If bucket is created than we add a default path expression. If customer provided bucket is used, then pass the customer provided path expression

Reply:
for both alb and CloudTrail